src/neo4j_genai/pipeline/types.py

```diff
@@ -0,0 +1,37 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+from pydantic import BaseModel, ConfigDict
+
+from neo4j_genai.pipeline.component import Component
+
+
+class ComponentDef(BaseModel):
```

Comment:
Minor comment but what does a def stand for in this PR?

Reply:
"Definition". Maybe "config" is clearer?

Reply:
Yeah I think config is clearer here. Wdyt?

Reply:
Totally agree, after saying it, it makes more sense. Done in https://github.com/neo4j/neo4j-genai-python/pull/81/commits/a6fb080fde19e5fb9b41e1b6603d67c399f8ddc2